### PR TITLE
chore: updates next/image uses to next/legacy/image

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task]()
🔍 [Preview Link](https://react-components-git-{branch-slug}-hashicorp.vercel.app)

---

## Description

Update react-components `next/image` uses to `next/legacy/image` for full compatibility with next 13 now that all of our sites have been upgraded.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [ ] Add Asana and Preview links above.
- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [ ] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203354752975570

### CODE REVIEW

File: .changeset/nasty-pans-hug.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+'@hashicorp/react-code-block': major
+'@hashicorp/react-docs-page': major
+'@hashicorp/react-hero': major
+'@hashicorp/react-author-byline': minor
+'@hashicorp/react-card': minor
+'@hashicorp/react-marketo-form': minor
+'@hashicorp/react-notification': minor
```
Comment (BRKalow) [inline .changeset/nasty-pans-hug.md:8]: The `minor` updates here are pre-1.0 packages, so this should be synonymous with a major per https://semver.org/#spec-item-4

> Version 1.0.0 defines the public API. The way in which the version number is incremented after this release is dependent on this public API and how it changes.

File: next-env.d.ts [hunk 2]
```diff
@@ -1,8 +1,3 @@
-/**
- * Copyright (c) HashiCorp, Inc.
- * SPDX-License-Identifier: MPL-2.0
- */
-
```
Comment (dstaley) [inline next-env.d.ts:5]: is the bot gonna get angry with us for removing this?

File: packages/card/primitives.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@
 
 import * as React from 'react'
 import classNames from 'classnames'
-import Image from 'next/image'
```
Comment (dstaley) [inline packages/card/primitives.tsx:8]: I think `<Card>` actually needs `next/image` since @alexcarpenter fixed the CSS.

**Review by dstaley — APPROVED** [review]
LGTM besides the `<Card>` which I believe works on `next/image`.

File: next-env.d.ts (hunk 2 above)
Comment (BRKalow) [reply next-env.d.ts:5]: Probably 👀 I think we can define some config to exclude this file, I'll do that in a follow up!

Comment (BRKalow) [reply next-env.d.ts:5]: https://github.com/hashicorp/react-components/pull/898

File: packages/card/primitives.tsx (hunk 3 above)
Comment (BRKalow) [reply packages/card/primitives.tsx:8]: Ah, great catch! I will revert this then 👍

**Pull request merged**